File: server/src/Core/Models/Course.cs
using Chad.Models.Common;

namespace Chad.Models
{
    public record Course : DescribedElementSummary
    {
        public UserSummary Director { get; init; }
        public ElementSummary[] Lessons { get; init; }
        public ElementSummary[] Classes { get; init; }
    }
}

File: client/src/app/models/course.ts
import {ElementSummary} from './common/element-summary';
import {DescribedElementSummary} from './common/described-element-summary';
import {UserSummary} from './user-summary';

export interface Course extends DescribedElementSummary {

  director: UserSummary;
  lessons: ElementSummary[];
  classes: ElementSummary[];
}
